File: src/main/java/dev/lonami/uniffidl/UdlFileType.java
package dev.lonami.uniffidl;

import com.intellij.openapi.fileTypes.LanguageFileType;
import com.intellij.openapi.util.NlsContexts;
import com.intellij.openapi.util.NlsSafe;
import org.jetbrains.annotations.NonNls;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import javax.swing.*;

public class UdlFileType extends LanguageFileType {
    public static final UdlFileType INSTANCE = new UdlFileType();

    private UdlFileType() {
        super(UdlLanguage.INSTANCE);
    }

    @Override
    public @NonNls @NotNull String getName() {
        return "UDL File";
    }

    @Override
    public @NotNull String getDescription() {
        return "UniFFI data language file";
    }

    @Override
    public @NotNull String getDefaultExtension() {
        return "udl";
    }

    @Override
    public @Nullable Icon getIcon() {
        return UdlIcons.FILE;
    }
}

File: src/main/kotlin/dev/lonami/uniffidl/UdlFileType.kt
package dev.lonami.uniffidl

import com.intellij.openapi.fileTypes.LanguageFileType
import org.jetbrains.annotations.NonNls
import javax.swing.Icon

class UdlFileType private constructor() : LanguageFileType(UdlLanguage.INSTANCE) {

    override fun getName(): @NonNls String = "UDL File"

    override fun getDescription(): String = "UniFFI data language file"

    override fun getDefaultExtension(): String = "udl"

    override fun getIcon(): Icon = UdlIcons.FILE

    companion object {
        @JvmStatic
        val INSTANCE = UdlFileType()
    }
}
